connector/src/test/java/com/google/cloud/spark/bigquery/OptimizeLoadUriListTest.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2020 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.spark.bigquery;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.google.common.truth.Truth.assertThat;
+
+@RunWith(Parameterized.class)
+public class OptimizeLoadUriListTest {
+
+  private List<String> input;
+  private List<String> expected;
+
+  public OptimizeLoadUriListTest(List<String> input, List<String> expected) {
+    this.input = input;
+    this.expected = expected;
+  }
+
+  @Parameterized.Parameters(name = "{index}: Should get {1}")
+  public static Iterable<Object[]> data() {
```

Comment:
Can we add edge cases for input list of size 0 and 1?

Reply:
Added, thanks